tests/Nhs.Appointments.Core.UnitTests/SiteServiceTests.cs

```diff
@@ -634,4 +634,76 @@ public async Task GetSiteByIdAsync_ReturnsDefault_WhenSiteIsNotFound(string scop
         var result = await _sut.GetSiteByIdAsync(siteId, scope);
         result.Should().BeNull();
     }
+
+    [Fact]
+    public async Task GetSitesPreview_CachedSitesFound_ReturningCachedSites()
+    {
+        var sites = new List<Site>()
+        {
+            new Site(
+                    Id: "ABC01",
+                    Name: "Site 1",
+                    Address: "1 Park Row",
+                    PhoneNumber: "0113 1111111",
+                    OdsCode: "odsCode1",
+                    Region: "R1",
+                    IntegratedCareBoard: "ICB1",
+                    Location: new Location(Type: "Point", Coordinates: [0.04, 50.0]),
+                    AttributeValues: new List<AttributeValue>() {new (Id: "accessibility/access_need_1", Value: "true")}),
+            new Site(
+                    Id: "ABC02",
+                    Name: "Site 2",
+                    Address: "2 Park Row",
+                    PhoneNumber: "0113 1111111",
+                    OdsCode: "odsCode2",
+                    Region: "R1",
+                    IntegratedCareBoard: "ICB1",
+                    Location: new Location(Type: "Point", Coordinates: [0.05, 50.0]),
+                    AttributeValues: new List<AttributeValue>() {new (Id: "accessibility/access_need_1", Value: "false")})
+        };
+        object? outSites = sites;
+        _memoryCache.Setup(x => x.TryGetValue("sites", out outSites)).Returns(true);
+        var result = await _sut.GetSitesPreview();
+
+        result.Count().Should().Be(2);
+        _siteStore.Verify(x => x.GetAllSites(), Times.Never);
+        _memoryCache.Verify(x => x.CreateEntry("sites"), Times.Never);
+    }
+
+    [Fact]
+    public async Task GetSitesPreview_CachedSitesNotFound_ReturningAllSitesFromSiteStore()
+    {
+        var sites = new List<Site>()
+        {
+            new Site(
+                    Id: "ABC01",
+                    Name: "Site 1",
+                    Address: "1 Park Row",
+                    PhoneNumber: "0113 1111111",
+                    OdsCode: "odsCode1",
+                    Region: "R1",
+                    IntegratedCareBoard: "ICB1",
+                    Location: new Location(Type: "Point", Coordinates: [0.04, 50.0]),
+                    AttributeValues: new List<AttributeValue>() {new (Id: "accessibility/access_need_1", Value: "true")}),
+            new Site(
+                    Id: "ABC02",
+                    Name: "Site 2",
+                    Address: "2 Park Row",
+                    PhoneNumber: "0113 1111111",
+                    OdsCode: "odsCode2",
+                    Region: "R1",
+                    IntegratedCareBoard: "ICB1",
+                    Location: new Location(Type: "Point", Coordinates: [0.05, 50.0]),
+                    AttributeValues: new List<AttributeValue>() {new (Id: "accessibility/access_need_1", Value: "false")})
+        };
+        object? outSites = null;
```

Comment:
This will generate build warnings because we've disabled Nullable types. I also notice there are 3 other uses already in this file that must've snuck in from a previous PR, would you mind just resolving them please? Just needs the `?` knocking off